src/lib.rs

```diff
@@ -179,24 +179,40 @@ impl<T> Cactus<T> {
         CactusValsIter{next: self.node.as_ref()}
     }
 
-    /// Consume this Cactus node and return its data. If the cactus node has no children, its data
-    /// is returned without cloning; if the node has children then the internal data is cloned
-    /// (hence why `T` must implement the `Clone` trait).
+    /// Try to consume this Cactus node and return its data. If the cactus node has no children,
+    /// this succeeds; if the cactus node has children, it fails, and returns the original
```

Comment:
Wouldn't it be better to return something more obviously an error? Result?

Reply:
This PR uses the API design from https://doc.rust-lang.org/std/rc/struct.Rc.html#method.try_unwrap as a guide.